src/plugins.ts

```diff
@@ -0,0 +1,82 @@
+import {PluginConfig, Pod} from './pod';
+import {join} from 'path';
+
+export default class Plugins {
+  static DefaultPluginsDir = 'plugins';
+  static DefaultPluginsFilename = 'plugin.js';
+
+  readonly pod: Pod;
+  readonly plugins: any[];
+
+  constructor(pod: Pod) {
+    this.pod = pod;
+    this.plugins = [];
+  }
+
+  /**
+   * Register plugins based on a plugin configuration. Normally from the amagaki.yaml file.
+   */
+  async registerPlugins(pluginConfigs?: Array<PluginConfig>) {
+    if (!pluginConfigs) {
+      return;
+    }
+
+    for (const pluginConfig of pluginConfigs) {
+      const localePluginPath = join(
+        Plugins.DefaultPluginsDir,
+        pluginConfig.key,
+        Plugins.DefaultPluginsFilename
+      );
+
+      // Check for local plugin definition.
+      if (this.pod.fileExists(localePluginPath)) {
+        const pluginImport = await import(
+          this.pod.getAbsoluteFilePath(localePluginPath)
+        );
+        const PluginClass =
+          pluginImport.plugin || pluginImport.default || pluginImport;
+
+        this.plugins.push(new PluginClass(this.pod, pluginConfig));
+        continue;
+      }
+
+      // TODO: Check for plugin in the node_modules.
```

Comment:
Hmm... maybe the import statement can be changed between local and installed plugins? Then we limit the amount of file path detection/management we do.

```
// Local
import('./foo.js');

// Installed
import('amagaki-plugin-foo')
```

Node should automatically look in the `node_modules` when the latter syntax is used.

Worth noting that if we move to the `amagaki.js` for plugins this is just "free" and we don't do anything special.

Reply:
That is a good idea to use just the package name for the packaged plugins.

Reply:
This is obsolete with the change over to having the `amagaki.js` to control configuration since the `amagaki.js` does the plugin `require` statements on their own.